src/parser/type_checking.c

```diff
@@ -18237,24 +18237,44 @@ pt_fold_const_expr (PARSER_CONTEXT * parser, PT_NODE * expr, void *arg)
     }
 
   if (prm_get_bool_value (PRM_ID_ORACLE_STYLE_EMPTY_STRING)
-      && (op == PT_STRCAT || op == PT_PLUS || op == PT_CONCAT || op == PT_CONCAT_WS))
+      && (op == PT_STRCAT || op == PT_PLUS || op == PT_CONCAT || op == PT_CONCAT_WS || op == PT_REPLACE))
     {
       TP_DOMAIN *domain;
 
+      if (op == PT_REPLACE)
+	{
+	  opd3 = expr->info.expr.arg3;
+	  type3 = (opd3) ? opd3->type_enum : PT_TYPE_NULL;
+	}
+
       /* use the caching variant of this function ! */
       domain = pt_xasl_node_to_domain (parser, expr);
 
       if (domain
 	  && (QSTR_IS_ANY_CHAR_OR_BIT (TP_DOMAIN_TYPE (domain)) || type1 == PT_TYPE_NULL || type2 == PT_TYPE_NULL))
 	{
-	  if (opd1 && opd1->node_type == PT_VALUE && type1 == PT_TYPE_NULL)
+	  if (opd3 && opd3->node_type == PT_VALUE && type3 == PT_TYPE_NULL)	/* REPLACE */
+	    {
+	      /* need to replace the NULL with empty string */
+	      int err =
+		db_string_make_empty_typed_string (&opd3->info.value.db_value, DB_TYPE_STRING, DB_DEFAULT_PRECISION,
+						   TP_DOMAIN_CODESET (domain), TP_DOMAIN_COLLATION (domain));
+	      if (err != NO_ERROR)
+		{
+		  has_error = true;
+		}
+	      else
+		{
+		  result = expr;
+		}
+	    }
+	  else if (opd1 && opd1->node_type == PT_VALUE && type1 == PT_TYPE_NULL)
```

Comment:
"select replace(null, 'a', 'A');" results in a wrong value 'a' due to this conditional block.
Separate handling of REPLACE seems necessary.

Reply:
Good point, I fixed it.